client/my-sites/checkout/composite-checkout/wpcom/hooks/use-shopping-cart.ts

```diff
@@ -103,6 +104,8 @@ function shoppingCartHookReducer(
 		case 'REMOVE_CART_ITEM': {
 			const uuidToRemove = action.uuidToRemove;
 			debug( 'removing item from cart with uuid', uuidToRemove );
+			removeItemFromLocalStorage( getProductSlug( state.responseCart, uuidToRemove ) );
```

Comment:
Sometimes removing an item from the cart results in multiple items being removed (eg: removing a domain when G Suite for that domain is also in the cart). In fact, we can rarely trust that any cart request will match the cart response. (This is why we can't do optimistic rendering of cart changes.) Will that cause any problems here? If so, maybe it would be better to recreate the local storage cart with the response when it comes in.

Reply:
I don't think it would be a problem in this case, since the signup flow now allows only a plan and domain to be added to cart, and we can have either of those products standalone if one of the items is removed from cart. However, I see your point and to future proof, I refactored the code to remove products from localstorage only after the response is received from cart.